src/infra/src/table/timed_annotations.rs

```diff
@@ -0,0 +1,481 @@
+    }
+    if let Some(text) = timed_annotation.text {
+        update_query = update_query.col_expr(timed_annotations::Column::Text, Expr::value(text));
+    }
+    if let Some(tags) = timed_annotation.tags {
+        // Serialize tags into a JSON value
+        let tags_json = serde_json::to_value(&tags).map_err(|e| {
+            let err_msg = format!("Failed to serialize tags: {}", e);
+            log::error!("{}", err_msg);
+            errors::Error::Message(err_msg)
+        })?;
+        update_query =
+            update_query.col_expr(timed_annotations::Column::Tags, Expr::value(tags_json));
+    }
+
+    // Step 4: Execute the update query
+    let result = update_query.exec(client).await?;
+
+    // Step 5: Check if any rows were affected
+    if result.rows_affected == 0 {
+        return Err(errors::Error::DbError(errors::DbError::KeyNotExists(
+            format!(
+                "TimedAnnotation with ID {} not found in dashboard {}",
+                timed_annotation_id, dashboard_id
+            ),
+        )));
+    }
+
+    Ok(())
+}
+
+pub async fn add(
+    dashboard_id: &str,
+    timed_annotation: TimedAnnotation,
+    use_given_id: bool,
+) -> Result<Vec<TimedAnnotation>, errors::Error> {
+    add_many(dashboard_id, vec![timed_annotation], use_given_id).await
+}
+
+pub async fn add_many(
+    dashboard_id: &str,
+    timed_annotations: Vec<TimedAnnotation>,
+    use_given_id: bool,
+) -> Result<Vec<TimedAnnotation>, errors::Error> {
+    // make sure only one client is writing to the database(only for sqlite)
+    let _lock = get_lock().await;
+
+    let client = ORM_CLIENT.get_or_init(connect_to_orm).await;
+
+    let txn = client.begin().await?;
+    let mut inserted_annotations = Vec::new();
+
+    for timed_annotation in timed_annotations {
+        let annotation =
+            insert_timed_annotation(&txn, dashboard_id, timed_annotation, use_given_id).await?;
+        inserted_annotations.push(annotation);
+    }
+
+    txn.commit().await?;
+    Ok(inserted_annotations)
+}
+
+/// Helper function to insert a single `TimedAnnotation` record and its associated panels
+async fn insert_timed_annotation<'a>(
+    txn: &'a DatabaseTransaction,
+    dashboard_id: &str,
+    timed_annotation: TimedAnnotation,
+    use_given_id: bool,
+) -> Result<TimedAnnotation, errors::Error> {
+    let dashboard_record = dashboards::Entity::find()
+        .filter(dashboards::Column::DashboardId.eq(dashboard_id))
+        .one(txn)
+        .await?
+        .ok_or_else(|| {
+            errors::Error::DbError(errors::DbError::KeyNotExists(format!(
+                "Dashboard with ID {} not found",
+                dashboard_id
+            )))
+        })?;
+    let dashboard_pk = dashboard_record.id;
+
+    let annotation_id: String = if use_given_id {
+        timed_annotation
+            .annotation_id
+            .unwrap_or_else(ider::uuid)
```

Comment:
We might want to rasie an error in this case rather than generate the KSUID. If the user set `use_given_id` to `true` but none was given, then this might indicate a problem in our supercluster code. So we might want to raise an error and discover the problem earlier rather than silently creating a new and inconsistent KSUID and only discovering the problem later.

Reply:
Yes, done.